### PR TITLE
Add `ssl_lib` and `ssl_inc` properties to `haproxy_install`

### PR DESCRIPTION
**Pull request by derekgroh**
### Description

Add `ssl_lib` and `ssl_inc` properties to `haproxy_install`

Addresses missing commits in #454, supersedes #462  

### Issues Resolved

None

### Contribution Check List

- [X] All tests pass.
- [X] New functionality includes testing.
- [X] New functionality has been documented in the README if applicable

### CODE REVIEW

**Comment by derekgroh** [comment]
Should additional platforms be covered in the CI?

`source-lua` was used as an example, but it seems the platform coverage is limited overall.

**Comment by bmhughes** [comment]
> 
> 
> Should additional platforms be covered in the CI?
> 
> `source-lua` was used as an example, but it seems the platform coverage is limited overall.

Yes I was going to suggest that, run it on as many platforms as possible. Use the source suites as an example to follow unless there's an overly hard to work around issue with a platform, in which case just exclude the problem platform(s).

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -101,6 +101,39 @@ jobs:
...
+          - "debian-10"
+          - "ubuntu-2004"
+          - "ubuntu-2104"
+          - "amazonlinux-2"
+          - "centos-7"
+          - "centos-8"
+          - "centos-stream-8"
+          - "fedora-latest"
+        suite:
+          - "source-openssl"
```
Comment (ramereth) [inline .github/workflows/ci.yml:120]: Instead of doing this here, it might make sense to just add the suite to the `dokken` job above.

**Review by ramereth — DISMISSED** [review]

File: documentation/haproxy_install.md [hunk 2]
```diff
@@ -28,25 +28,27 @@ This resource also uses the following partial resources:
...
+| `use_zlib`           | String      | `true`                                                           | Include ZLIB support                                                           | `true`, `false`     |
+| `use_linux_tproxy`   | String      | `true`                                                           |                                                                                | `true`, `false`     |
+| `use_linux_splice`   | String      | `true`                                                           |                                                                                | `true`, `false`     |
+| `use_systemd`        | String      | `true`                                                           |                                                                                | `true`, `false`     |
+| `use_lua`            | String      | `false`                                                          | Include Lua support                                                            | `true`, `false`     |
```
Comment (ramereth) [inline documentation/haproxy_install.md:47]: These are not `String` but are `Boolean`

**Comment by derekgroh** [comment]
Feel like I'm going in circles trying to get an OS exclude for `fedora-latest`. Anyone know how best to do so?

https://docs.github.com/en/actions/reference/workflow-syntax-for-github-actions#example-excluding-configurations-from-a-matrix
Seems to document the format as strings in brackets without quotes, but the checks failed to either value.

**Comment by bmhughes** [comment]
> 
> 
> Feel like I'm going in circles trying to get an OS exclude for `fedora-latest`. Anyone know how best to do so?
> 
> https://docs.github.com/en/actions/reference/workflow-syntax-for-github-actions#example-excluding-configurations-from-a-matrix
> Seems to document the format as strings in brackets without quotes, but the checks failed to either value.

Looks correct to me and the latest action runs has excluded as configured? Unless I'm missing something?

**Comment by derekgroh** [comment]
Guess GitHub Actions was broken at the time, kept getting failures. Guess the question is which format is best?

Assume ffb1e74 would be best

**Review by ramereth — APPROVED** [review]

**Comment by kitchen-porter** [comment]
Released as: [12.1.0](https://github.com/sous-chefs/haproxy/releases/tag/12.1.0)

**Pull request merged**